src/main/java/gov/usgs/wma/waterdata/groundwater/DiscreteGroundWaterRules.java

```diff
@@ -0,0 +1,87 @@
+package gov.usgs.wma.waterdata.groundwater;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import org.springframework.util.StringUtils;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+/**
+ * Applies business rules to a domain object.
+ * It is assumed that this object can be modified.
+ */
+public class DiscreteGroundWaterRules {
+
+	//Threadsafe factory
+	private static final JsonFactory jsonFactory = new JsonFactory();
+
+	/**
+	 * Apply business rules to a DiscreteGroundWater, modifying it in place.
+	 * @param domObj
+	 */
+	public void apply(DiscreteGroundWater domObj) {
+
+		//Rule:  The AQ measurement reading qualifiers (multi-values) are mapped to a single NWISWeb lev_status_cd.
+		//If the AQ qualifier(s) are not recognized as valid AQ qualifiers, it is ignored.
+		//If the AQ qualifier(s) are empty or just not one that is mapped, it is assumed to be an unqualified measurement.
+		//Ref:  https://internal.cida.usgs.gov/jira/browse/IOW-667
+		//
+		//There is no order or preference among the multi-valued AQ qualifiers to select a single one for NWISWeb.
+		//To be repeatable, it is assumed that the qualifiers will be matched in the order of the LevelStatusCode enum,
+		//not in order of the values in the db column (order is not repeatable within a JSONB object).
+		{
+			String orgQualStr = StringUtils.trimWhitespace(domObj.readingQualifiers);
+			String newQualStr = "";   //acceptable default value if no other found
+
+			if (! StringUtils.isEmpty(orgQualStr)) {
+				try {
+
+					List<String> aqQuals = new ArrayList();
+
+					JsonParser parser = jsonFactory.createParser(orgQualStr);
+
+					while (! parser.isClosed()) {
+
+						JsonToken token = parser.nextToken();
+
+						if (token != null && token.isScalarValue()) {
+							String val = StringUtils.trimWhitespace(parser.getValueAsString());
+
+							if (!StringUtils.isEmpty(val)) {
+								aqQuals.add(StringUtils.trimWhitespace(parser.getValueAsString()));
+							}
+						}
+					}
+
+					if (aqQuals.size() > 0) {
+
+						LevelStatusCode lsc = Stream.of(LevelStatusCode.values())
+													  .filter(a -> a.isMapped())
+								                      .filter(a -> aqQuals.contains(a.getAqDescription()))
+								                      .findFirst().orElse(null);
+
+						if (lsc != null) {
+							newQualStr = lsc.getNwisCode();
+						}
+
+					}
+
+				} catch (IOException e) {
+					throw new RuntimeException(e);
+				}
+
+			}
+
+			domObj.readingQualifiers = newQualStr;
```

Comment:
It is not clear to me why this code goes through enum gyrations to only return a string in the end. I think it should return an enum, no?

Reply:
Yeah, good question, and one I think is worth discussing b/c it gets a the basic pattern of how we ETL values.
What we get from AQTS is a JSON String containing an array of values.  What we need to put into the RDB is a single digit.  The incoming JSON String might be a bit negotiable, but that is currently what is preserved all the way from AQTS into the Observations DB.

One simple way to think of this is that the enum is a stand-in for a DAO.  For now the list is considered unchanging, but the enum could be thought of as just a structured hack to hardcode what might have been a db look-up.

I want to go a little further, though, because there are cases where we use enums for simple, unchanging value lists.  First, my key goal in these ETLs is to separate and consolidate the Business Logic for several reasons:
* By separating BL from other code, I can more easily unit test it without having to do IT tests w/ db CSV files to create edge cases.  Unit tests let me really pound the BL w/ every weird edge case I can think of.
* By consolidating the BL, it's much easier to review what rules (validations, mappings, calculations) are in place.

With that in mind, I have a Rules object that processes a Domain object applying the BL rules.  The dom object needs to be able to hold the AQTS value (String) and a representation of the output value.  If I want to put an enum into the dom obj, I could:
* Add a separate field to hold the enum or
* I could have two dom objs (separate AQTS and NWISWeb ones)
I think both options are more complex than needed.  My source value and destination value are both Strings, so it works to just modify the existing value according to the rules.

I also think that it is slightly preferable to not put an enum in the dom obj when following this 'rules' pattern, since it means that this field becomes special, requiring some special code to dereference the enum to its string representation.  That results in just a little bit of logic when we write from the dom obj.

Reply:
I think you explained the part I understand and not the part I am unclear about. We discussed the enum mechanism together  as a DOA substitute. What I am unclear about is that an enum is not returned once it is determined. Why a string?